src/erlcloud_lambda.erl

```diff
@@ -377,9 +377,10 @@ invoke(FunctionName, Payload, Options, Qualifier) when is_binary(Qualifier) ->
              Qualifier :: binary()| undefined,
              Config    :: aws_config()) -> return_val().
 invoke(FunctionName, Payload, Options, Qualifier, Config = #aws_config{}) ->
-    Path = base_path() ++ "functions/" ++ binary_to_list(FunctionName) ++ "/invocations",
+    URLFunctionName = erlcloud_http:url_encode(binary_to_list(FunctionName)),
```

Comment:
I just did this here for now, @motobob @wvj384 can you check that this works for you? It looks like it must be the issue from [eate a canonical request for Signature Version 4](https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html).

If that is right, I will fix all the other URL generations in this module, it looks like they're all potentially impacted if ARN is passed as function name.

Reply:
does not seem so.

Reply:
The secret was double URL encoding. According to that signing document, which I missed at first, that is what it expects.

> Each path segment must be URI-encoded ***twice*** (except for Amazon S3 which only gets URI-encoded once).
> Example Canonical URI with encoding
> `/documents%2520and%2520settings/`

So, we URL-encode all parameters in the URL everywhere in this module, then loose-URL-encode the URI when passing to sign_v4.

I don't know if it would be wise to do this generally in other APIs, but it sure isn't centralized now.